File: construct-quad-tree/Node.kt
package com.github.masx200.leetcode_test.construct_quad_tree


class Node(
    var `val`: Boolean, var isLeaf: Boolean, var topLeft: Node? = null,
    var topRight: Node? = null,
    var bottomLeft: Node? = null,
    var bottomRight: Node? = null
)

File: construct-quad-tree/Node.ts
export default class Node {
    val: boolean;
    isLeaf: boolean;
    topLeft: Node | null;
    topRight: Node | null;
    bottomLeft: Node | null;
    bottomRight: Node | null;
    constructor(
        val?: boolean,
        isLeaf?: boolean,
        topLeft?: Node | null,
        topRight?: Node | null,
        bottomLeft?: Node | null,
        bottomRight?: Node | null,
    ) {
        this.val = val === undefined ? false : val;
        this.isLeaf = isLeaf === undefined ? false : isLeaf;
        this.topLeft = topLeft === undefined ? null : topLeft;
        this.topRight = topRight === undefined ? null : topRight;
        this.bottomLeft = bottomLeft === undefined ? null : bottomLeft;
        this.bottomRight = bottomRight === undefined ? null : bottomRight;
    }
}
